frontend/src/features/admin-form/responses/ChartsPage/UnlockedCharts/UnlockedChartsContainer.tsx

```diff
@@ -0,0 +1,224 @@
+        hashMap.set(field.answer, (hashMap.get(field.answer) || 0) + 1)
+      } else if (field._id === id && field.answerArray) {
+        // multi answer fields, like checkboxes
+        field.answerArray.forEach((answer) => {
+          if (typeof answer === 'string')
+            return hashMap.set(answer, (hashMap.get(answer) || 0) + 1)
+        })
+      }
+    })
+  })
+
+  return Array.from(hashMap)
+}
+
+// transform filtered text data into an array of {word: count}
+const aggregateWordCloud = (
+  id: string,
+  data: DecryptedSubmission[],
+): WordCloudProps['words'] => {
+  const hashMap = new Map<string, number>()
+
+  const resultArr: WordCloudProps['words'] = []
+
+  data.forEach((content) => {
+    content.responses.forEach((field) => {
+      if (field._id === id && field.answer) {
+        // split to words
+        const answerArray = field.answer.split(' ')
+        // remove stop words from array
+        const ansNoStopW = removeStopwords(answerArray)
+        ansNoStopW.forEach((word) => {
+          // remove punctuations
+          const wordNoPunc = word.replace(/\W|_/g, '')
+          // normalise to lower case
+          const wordLower = wordNoPunc.toLowerCase()
+          hashMap.set(wordLower, (hashMap.get(wordLower) || 0) + 1)
+        })
+      }
+    })
+  })
+  hashMap.forEach((val, key) => resultArr.push({ text: key, value: val }))
+  return resultArr
+}
+
+export const UnlockedChartsContainer = () => {
+  const { data: decryptedContent } = useAllSubmissionData()
+  const { data: form } = useAdminForm()
+  const { dateRange, setDateRange } = useStorageResponsesContext()
+
+  const filteredDecryptedData: DecryptedSubmission[] = useMemo(() => {
+    if (!decryptedContent) return []
+    if (dateRange.length === 2) {
+      const [startDate, endDate] = dateRange.map((date) =>
+        new Date(date).getTime(),
+      )
+      // Set to the end of the endDate
+      const endOfDay = addDays(endDate, 1).getTime() - 1
+
+      return decryptedContent.filter((content) => {
+        const submissionTime = new Date(content.submissionTime).getTime()
+        return submissionTime >= startDate && submissionTime <= endOfDay
+      })
+    }
+    return decryptedContent
+  }, [decryptedContent, dateRange])
+
+  const prettifiedResponsesCount = useMemo(
+    () =>
+      dateRange.length === 2
+        ? simplur` ${[filteredDecryptedData.length ?? 0]}result[|s] found`
+        : simplur` ${[filteredDecryptedData.length ?? 0]}response[|s] to date`,
+    [filteredDecryptedData, dateRange],
+  )
+
+  if (!form) return null
+
+  const renderedCharts = form.form_fields
+    .map((formField, idx) => {
+      const questionTitle = `${idx + 1}. ${formField.title}`
+
+      // if field type is text, create word cloud
+      if (
+        formField.fieldType === BasicField.ShortText ||
+        formField.fieldType === BasicField.LongText
+      ) {
+        const words = aggregateWordCloud(formField._id, filteredDecryptedData)
```

Comment:
issue:
- if i delete a form field, responses corresponding to that form field id will never show up in chart again
- it is not possible to map form fields based on responses (instead of form def) because responses do not contain field type